src/qiskit_sphinx_theme/assets/scripts/qiskit-sphinx-theme.js

```diff
@@ -1,4 +1,181 @@
+    // // scroll the currently "active" heading in toc, into view.
+    // const rect = current.getBoundingClientRect();
+    // if (0 > rect.top) {
+    //   current.scrollIntoView(true); // the argument is "alignTop"
+    // } else if (rect.bottom > window.innerHeight) {
+    //   current.scrollIntoView(false);
+    // }
+  }
+}
+
+function scrollHandler(positionY) {
+  scrollHandlerForHeader();
+  scrollHandlerForBackToTop(positionY);
+  scrollHandlerForTOC(positionY);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Theme Toggle
+////////////////////////////////////////////////////////////////////////////////
+function setTheme(mode) {
+  if (mode !== "light" && mode !== "dark" && mode !== "auto") {
+    console.error(`Got invalid theme mode: ${mode}. Resetting to auto.`);
+    mode = "auto";
+  }
+
+  document.body.dataset.theme = mode;
+  localStorage.setItem("theme", mode);
+  console.log(`Changed to ${mode} mode.`);
+}
+
+function cycleThemeOnce() {
+  const currentTheme = localStorage.getItem("theme") || "auto";
+  const prefersDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
+
+  if (prefersDark) {
+    // Auto (dark) -> Light -> Dark
+    if (currentTheme === "auto") {
+      setTheme("light");
+    } else if (currentTheme == "light") {
+      setTheme("dark");
+    } else {
+      setTheme("auto");
+    }
+  } else {
+    // Auto (light) -> Dark -> Light
+    if (currentTheme === "auto") {
+      setTheme("dark");
+    } else if (currentTheme == "dark") {
+      setTheme("light");
+    } else {
+      setTheme("auto");
+    }
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Setup
+////////////////////////////////////////////////////////////////////////////////
+function setupScrollHandler() {
+  // Taken from https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event
+  let last_known_scroll_position = 0;
+  let ticking = false;
+
+  window.addEventListener("scroll", function (e) {
+    last_known_scroll_position = window.scrollY;
+
+    if (!ticking) {
+      window.requestAnimationFrame(function () {
+        scrollHandler(last_known_scroll_position);
+        ticking = false;
+      });
+
+      ticking = true;
+    }
+  });
+  window.scroll();
+}
+
+function setupScrollSpy() {
+  if (tocScroll === null) {
+    return;
+  }
+
+  // Scrollspy -- highlight table on contents, based on scroll
+  new Gumshoe(".toc-tree a", {
+    reflow: true,
+    recursive: true,
+    navClass: "scroll-current",
+    offset: () => {
+      let rem = parseFloat(getComputedStyle(document.documentElement).fontSize);
+      // QISKIT CHANGE: start. Add 3.5rem for the Qiskit top nav bar.
+      // See _top-nav-bar.scss for where the value comes from.
+      return header.getBoundingClientRect().height + (0.5 * rem) + 1 + (3.5 * rem);
+      // QISKIT CHANGE: end.
```

Comment:
This is where the change happens

Reply:
Is it possible to export the `3.5rem` from the scss file and import it in this file? (As shown in [this blog post](https://www.bluematador.com/blog/how-to-share-variables-between-js-and-sass))

If we can just add and `export` and `import` line, then we should do it. If it's more involved then leave it.

Reply:
I tried out your suggestion and it's non-trivial because we're not using CSS modules and we combine all the SCSS into a single file. Given how stable this value is, I don't think it's worth it.

But thank you for the suggestion! TIL